homeassistant/components/matter/climate.py

```diff
@@ -134,29 +135,49 @@ def target_temperature(self) -> float | None:
             if value := self.get_matter_attribute_value(
                 clusters.Thermostat.Attributes.OccupiedCoolingSetpoint
             ):
+                assert isinstance(value, float)
                 return value / 100
         if value := self.get_matter_attribute_value(
             clusters.Thermostat.Attributes.OccupiedHeatingSetpoint
         ):
+            assert isinstance(value, float)
             return value / 100
         return None
 
-    async def async_set_temperature(self, **kwargs) -> None:
+    async def async_set_temperature(self, **kwargs: Any) -> None:
         """Set new target temperature."""
         target_temp: float | None = kwargs.get(ATTR_TEMPERATURE)
-        if target_temp is None:
+
+        if target_temp is None or self.target_temperature is None:
             return
+
         temp_diff = int((target_temp - self.target_temperature) * 10)
-        command = clusters.Thermostat.Commands.SetpointRaiseLower(2, temp_diff)
+
+        match self.hvac_mode:
+            case HVACMode.HEAT:
+                command = clusters.Thermostat.Commands.SetpointRaiseLower(
+                    clusters.Thermostat.Enums.SetpointAdjustMode.kHeatSetpoint,
+                    temp_diff,
+                )
+            case HVACMode.COOL:
+                command = clusters.Thermostat.Commands.SetpointRaiseLower(
+                    clusters.Thermostat.Enums.SetpointAdjustMode.kCoolSetpoint,
+                    temp_diff,
+                )
+            case HVACMode.AUTO:
```

Comment:
The auto mode doesn't support manual temperatur control. The heat_cool mode supports both low and high setpoints.

Reply:
In ha, are auto mode and heat_cool mode the same?

Reply:
No. Auto mode means the user can't set the target temperature, while heat_cool mode allows the user to set low and high target temperature setpoints.

Both modes have some level of automatic control, but they are not the same.

https://developers.home-assistant.io/docs/core/entity/climate#hvac-modes

Reply:
I couldn't find the set methods for low and high in the base class of climate https://github.com/home-assistant/core/blob/dev/homeassistant/components/climate/__init__.py#L402

Reply:
We use `async_set_temperature` to set those setpoints as well. Just look for the proper parameter names.

Here's the service schema.

https://github.com/home-assistant/core/blob/48049d588cf394bd9bace94206aae2934b43ceff/homeassistant/components/climate/__init__.py#L118-L130

Reply:
Here we may need to wait for the write attribute of the matter server to be added to the dependency of HA before we can proceed, because the setpoint cannot be achieved through a command

Reply:
We actually added that yesterday based on your feedback so we could move on and add that in this beta too.
So if I understand it correctly, some thermostats do support setting the hvac_mode but you need to write the attribute yourself instead that there's is some ready made command within the cluster, right ?

Reply:
As soon as [this PR](https://github.com/home-assistant/core/pull/95519) is merged, you can rebase your branch and you will have a "write_attribute" available in the client.